slimta/util/typecheck.py

```diff
@@ -0,0 +1,9 @@
+def check_argtype(val, type_, name, or_none=False):
```

Comment:
What is the purpose of this function?

It might be naive of me, but like I mentioned before about duck-typing I usually prefer to let functions "fail on their own" when given bad arguments, not force a failure by type checking. For example, in Python 3 a function that expects a str and gets a bytestring will probably fail if `.encode()` is called on it. In Python 2, it probably won't fail, but it will just be redundant.